fs/span-manager/span_manager.go

```diff
@@ -88,12 +89,23 @@ type spanInfo struct {
 
 // New creates a SpanManager with given ztoc and content reader, and builds all
 // spans based on the ztoc.
+
+// Should this return errors/nil objects on failure?
```

Comment:
Kind of. I attempted to do this but it ended up breaking [TestSpanManager](https://github.com/awslabs/soci-snapshotter/blob/cf3f3d9874b22a3b9efd0c777bb56f6914ceef96/fs/span-manager/span_manager_test.go#L33) case #3 (span digest verification fails), because it expects this malformed SpanManager to then return an incorrect value in other function calls.

I guess I could probably slim that down and ship it in this change, though.

Reply:
This seems to take more work than I thought. Will leave this out of the PR. In the meantime opened #1172 to make sure we get back to this.